pkg/globalmanager/runtime/worker.go

```diff
@@ -269,17 +290,43 @@ func injectDeploymentParam(deployment *appsv1.Deployment, workerParam *WorkerPar
 	if deployment.Spec.Template.Labels == nil {
 		deployment.Spec.Template.Labels = make(map[string]string)
 	}
+	if deployment.Spec.Selector.MatchLabels == nil {
+		deployment.Spec.Selector.MatchLabels = make(map[string]string)
+	}
 
 	for k, v := range generateLabels(object, workerParam.WorkerType) {
 		deployment.Labels[k] = v
 		deployment.Spec.Template.Labels[k] = v
 		deployment.Spec.Selector.MatchLabels[k] = v
 	}
-	deployment.Spec.Template.Spec.Containers[0].Ports = []v1.ContainerPort{
-		{
-			ContainerPort: port,
-		},
+
+	// Edgemesh part, useful for service mapping
+	deployment.Labels["app"] = object.GetName() + "-" + workerParam.WorkerType + "-" + "svc"
+	deployment.Spec.Template.Labels["app"] = object.GetName() + "-" + workerParam.WorkerType + "-" + "svc"
+	deployment.Spec.Selector.MatchLabels["app"] = object.GetName() + "-" + workerParam.WorkerType + "-" + "svc"
```

Comment:
`appLabelVaule := object.GetName() + "-" + workerParam.WorkerType + "-" + "svc"` since it occurs three times.